src/definition.jl

```diff
@@ -98,52 +94,62 @@ function handle_DefinitionRequest(server::Server, msg::DefinitionRequest)
 
     st0 = JS.build_tree(JL.SyntaxTree, fi.parsed_stream)
     offset = xy_to_offset(fi, origin_position)
-    node = select_target_node(st0, offset)
-    if node === nothing
-        return send(server, DefinitionResponse(; id = msg.id, result = null))
-    end
 
     locationlink_support = supports(server, :textDocument, :definition, :linkSupport)
-    originSelectionRange = get_source_range(node)
 
-    lbs = definition_target_localbindings(offset, st0, node, uri)
-    if !isnothing(lbs)
+    target_binding_definitions = local_definitions(st0, offset)
+    if !isnothing(target_binding_definitions)
+        target_binding, definitions = target_binding_definitions
+        local result = Location[
+            Location(; uri, range = get_source_range(definition))
+            for definition in definitions]
+        if locationlink_support
+            result = LocationLink[
+                LocationLink(loc, get_source_range(target_binding))
+                for loc in result]
+        end
         return send(server,
             DefinitionResponse(;
                 id = msg.id,
-                result = create_definition.(lbs,
-                    Ref(originSelectionRange),
-                    Ref(locationlink_support))))
+                result))
     end
 
+    node = select_target_node(st0, offset)
+    if node === nothing
+        return send(server, DefinitionResponse(; id = msg.id, result = null))
+    end
     (; mod, analyzer) = get_context_info(server.state, uri, origin_position)
     objtyp = resolve_type(analyzer, mod, node)
-
     objtyp isa Core.Const || return send(server, DefinitionResponse(; id = msg.id, result = null))
 
     objval = objtyp.val
+    originSelectionRange = get_source_range(node)
     if objval isa Module
         if is_location_unknown(objval)
             return send(server, DefinitionResponse(; id = msg.id, result = null))
         else
+            local result = Location(objval)
+            if locationlink_support
+                result = LocationLink(result, originSelectionRange)
```

Comment:
Surprisingly, it seems that they do not accept a single LocationLink as a valid response 😇

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_definition:~:text=DefinitionParams%20extends%20TextDocumentPositionParams%2C%0A%09WorkDoneProgressParams%2C%20PartialResultParams%20%7B%0A%7D-,Response%3A,-result%3A%20Location%20%7C%20Location%5B%5D%20%7C%20LocationLink%5B%5D%20%7C%20null

Reply:
Good catch!